pkg/controllers/resources/runtimeclasses/syncer.go

```diff
@@ -47,6 +49,11 @@ func (i *runtimeClassSyncer) Syncer() syncertypes.Sync[client.Object] {
 }
 
 func (i *runtimeClassSyncer) SyncToVirtual(ctx *synccontext.SyncContext, event *synccontext.SyncToVirtualEvent[*nodev1.RuntimeClass]) (ctrl.Result, error) {
+	if !ctx.Config.Sync.FromHost.RuntimeClasses.Selector.Matches(event.Host) {
+		i.EventRecorder().Eventf(event.Host, "Warning", "SyncWarning", "did not sync runtime class %q to virtual because it does not match the selector under 'sync.fromHost.runtimeClasses.selector'", event.Host.Name)
```

Comment:
We cannot use event.Host here as the event recorder only works for the virtual object